legend-engine-language-pure-store-relational/src/main/java/org/finos/legend/engine/language/pure/grammar/from/RelationalParseTreeWalker.java

```diff
@@ -1098,7 +1098,7 @@ private PropertyMapping visitPropertyMappingWithLocalMappingProperty(RelationalP
         multiplicity.setUpperBound("*".equals(toMultiplicityContext.getText()) ? null : Integer.parseInt(toMultiplicityContext.getText()));
         PropertyPointer propertyPointer = new PropertyPointer();
         propertyPointer.property = PureGrammarParserUtility.fromIdentifier(ctx.identifier());
-        propertyPointer._class = localMappingPropertyInfo.type;
+        propertyPointer._class = _class;
```

Comment:
What does this fix? Ideally, we don't want to infer anything in the grammar as it should be 1 to 1 with the protocol. Could we fix the compiler instead?

Reply:
As of now we are assigning type or property as its owner which is wrong. We can do this at parser stage because looking at the class mapping we know the name of property owner. We will resolve the owner at the compilation stage